go/pkg/client/client.go

```diff
@@ -474,6 +474,12 @@ func getRPCCreds(ctx context.Context, credFile string, useApplicationDefault boo
 	return rpcCreds, CredsFileAuth, nil
 }
 
+// ByteStreamWriteOpts sets options required for resumable ByteStream write operations.
+type ByteStreamWriteOpts struct {
```

Comment:
Have you considered using [variadic options](https://google.github.io/styleguide/go/best-practices#variadic-options)? It allows you to reuse the existing method without affecting existing call sites.

Reply:
Thanks for the input. I have considered to use it but adding my changes to the current existing method(WriteBytes in bytestream.go) seems making readability not great. Would you recommend me to use variadic options over an option struct?

Reply:
The new method shares most of its code with the existing one, which seems like a good candidate for an option parameter. Variadic options allow backwards compatibility with existing call sites.

I'm thinking something like:
```go
type ByteStreamWriteOption interface {
  Apply(*ByteStreamWriteOpts)
}
type ByteStreamOptFinishWrite bool
func (o ByteSteramOptFinishWrite) Apply(opts *ByteStreamWriteOpts) {
  opts.FinishWrite = bool(o)
}

func (c *Client) WriteBytesWithOffset(ctx context.Context, name string, data []byte, rawOpts ...ByteStreamWriteOption) (int64, error) {
  opts := &ByteStreamWriteOpts{}
  for _, o := range rawOpts {
    o.Apply(opts)
  }
  ...
}
```

Reply:
Update the code per your suggestion. Please let me know if I'm missing something.

Reply:
My bad, I meant to write the original public method name `WriteBytes` rather than add another version:
```go
func (c *Client) WriteBytes(ctx context.Context, name string, data []byte, rawOpts ...ByteStreamWriteOption) (int64, error) {
  ...
}